src/main/groovy/org/owasp/dependencycheck/gradle/tasks/ConfiguredTask.groovy

```diff
@@ -150,6 +150,27 @@ abstract class ConfiguredTask extends DefaultTask {
         settings.setBooleanIfNotNull(ANALYZER_OSSINDEX_USE_CACHE, config.cache.ossIndex)
     }
 
+    private void configureProxy(Settings settings) {
+        HttpProxySettings proxyGradle = new JavaSystemPropertiesSecureHttpProxySettings()
+        if (proxyGradle.proxy == null) {  // if systemProp.https.proxyHost is not defined, fallback to http proxy
+            proxyGradle = new JavaSystemPropertiesHttpProxySettings()
+        }
+
+        if (proxyGradle.proxy != null) {
+            config.proxy.server = proxyGradle.proxy.host
+            config.proxy.port = proxyGradle.proxy.port
+            config.proxy.username = proxyGradle.proxy.credentials.username
+            pconfig.roxy.password = proxyGradle.proxy.credentials.password
+            config.proxy.nonProxyHosts = getNonProxyHosts(proxyGradle)
+
```

Comment:
To make the deprecation of the `ProxyExtension` we should end the if block starting on line 159 here. That way if someone did have a proxy configured using the deprecated method it would still work.

Reply:
Yes: better transition
No: if you are behind a proxy like me, your gradle proxy is already set